collections/_posts/2022-10-26-GeneralizationInFederatedLearning.md

```diff
@@ -0,0 +1,102 @@
+* Propose a **three-way split** for measuring out-of-sample and participation gaps in centralized and FL settings where data is drawn from *a distribution of distributions*,
+* Observe **significant participation gaps** across different tasks and perform empirical studies on how number of clients, client diversity, etc. affect generalization performance,
+* Observe **significant differences in generalization behavior** across *naturally-partitioned* and *synthetically-partitioned* federated datasets,
+* Propose **semantic partitioning** as a synthetic partitioning principle.
+
+
+# Definition of a federated learning problem
+To correctly present the few theoretical ideas of the paper, we rewrite the notations and hypothesis of the authors.
+
+* *Let $$ \Xi $$ be the collection of all the possible image-label pairs,*
+* *For any parameters $$w$$ in a parameter space $$\Theta$$, $$f(w,\xi)$$ the loss at element $$\xi\in\Xi$$ with parameter $$w$$,*
+* *Let $$C$$ be the collection of all the possible clients. A client $$c\in C$$ is associated with a local distribution $$D_c$$ supported on $$\Xi$$,*
+* *Assume there is a meta-distribution $$P$$ supported on client set $$C$$, and each client $$c$$ is associated with a weight $$\rho_c$$ for aggregation.*
+
+**The proposed formulation of a federated learning problem in this setup is to optimize**
+
+$$ F(w) = F_{unpart}(w) := \mathbb{E}_{c\sim P}\left[\rho_c*\mathbb{E}_{\xi\sim D_c}[f(w,\xi)]\right] $$
+
+The formulation is more justified in a cross-device setting (with a large amount of clients, e.g. local distributions).
+
+# The empirical risks and generalization gaps
+## Definition
+* *Let $$\hat{C}$$ be the participating clients, drawn from the meta-distribution $$P$$,*
+* *For each $$c\in\hat{C}$$, let $$\hat{\Xi}_c$$ be the participating training client data, drawn from the local distribution $$D_c$$.*
+
+**They further define the *empirical risk* on the participating training client data with**
+
+$$ F_{part\_train}(w) := \frac{1}{|\hat{C}|}\underset{c\in \hat{C}}{\sum}[\rho_c*(\frac{1}{|\hat{\Xi}_c|}\underset{\xi\in\hat{\Xi}_c}{\sum} f(w,\xi))]$$
+
+**And the *semi-empirical risk* on the participating validation client data with**
+
+$$ F_{part\_val}(w) := \frac{1}{|\hat{C}|}\underset{c\in\hat{C}}{\sum}[\rho_c*(\mathbb{E}_{\xi\sim D_c}f(w,\xi))]$$
+
+They finally define two levels of generalization gap:
+* **The out-of-sample gap: $$ F_{part\_val}(w) - F_{part\_train}(w)$$**
+* **The partipation gap: $$ F_{unpart}(w) - F_{part\_val}(w)$$**
+
+## Highlighted properties
+* *Participation gap is an intrinsic property of FL due to heterogeneity*: the gap will not exist if data is homogeneous,
+* *Participation gap can quantify client diversity* (comparable between tasks),
+* *Participation gap can measure overfitting on the population distribution*,
+* *Participation gap can quantify model robustness to unseen clients*,
+* *Participation gap can quantify the incentive for clients to participate*.
```

Comment:
I'd reformat this to avoid repeating _participation gap_ so many times the following way :wink:
```suggestion
The participation gap:
* *is an intrinsic property of FL due to heterogeneity*. The gap will not exist if data is homogeneous,
* *can quantify client diversity* (comparable between tasks),
* *can measure overfitting on the population distribution*,
* *can quantify model robustness to unseen clients*,
* *can quantify the incentive for clients to participate*.
```